test/unit/ajax.js

```diff
@@ -1410,6 +1410,22 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - don't escape %20 in contentType (gh-4119)", 1, function( assert ) {
```

Comment:
This is testing one intended side-effect of the added prefilter but we should also test it more explicitly. Please also add tests checking that the `contentType` field gets correctly overwritten.

Reply:
I struggled with implementing that. I cannot use beforeSend since that seems to be called prior to the prefilter. The XHR object (and also jqXHR) does not give me read access to the request headers sent. Do you have any ideas or pointers for me?

Reply:
@wenz Are you sure? It should be called after the prefilter.

Reply:
Here's the test I tried within an `ajaxTest()` call, which always fails (s.contentType is "application/json"):

```
return {
	url: "bogus.html",
	contentType: "application/json",
	headers: { "content-type": "application/x-www-form-urlencoded" },
	beforeSend: function( _, s ) {
		assert.strictEqual( s.contentType, "application/x-www-form-urlencoded", "contentType is overwritten" );
		return false;
	},
	error: true
};
```
Is there something obvious I'm doing wrong?

The %20 tests indirectly check whether the prefilter works, so at least we have coverage of that code (?).

Reply:
@wenz That test passes for me on your branch. How are you running the tests? You need to either run `grunt` first or check the `esmodules` checkbox which will make tests use jQuery source files directly instead of the built file.

Reply:
I did run `grunt`, but most probably I messed up somewhere else. I just tried from scratch, and indeed now the test passes. Thank you for your patience! :)